enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,701 @@
+reported to the CRD and no BGP configuration will be done by OVN-Kubernetes.
+
+The CRD will support enabling advertisements for pod subnet and egress IPs. Note, MetalLB still handles advertising
+LoadBalancer IP so there is no conflict of responsibilities here. When the pod network is set to be advertised, there is
+no longer a need to SNAT pod IPs for this network to the node IP. Therefore, when pod network advertisements are enabled,
+the traffic from these pods will no longer be SNAT'ed on egress.
+
+The "targetVRF" key is used to determine which VRF the routes should be advertised in. The default value is "auto", in
+which case OVN-Kubernetes will advertise routes in the VRF that corresponds to the selected network. Alternatively, the
+user may specify the name of the VRF, which would cause routes to be leaked for this network onto that VRF. One use case
+for this would be when a user wants to define a network with a specific IP addressing scheme, and then wants to advertise
+the pod IPs into the provider BGP network without VPN. By specifying the targetVRF as "default", routes will be leaked
+into the default VRF. Note that by using route leaking with a user-defined network, the network is no longer fully
+isolated, as now any other networks also leaked or attached to that VRF may reach this user-defined network. If a user
+attempts to leak routes into a targetVRF for a user-defined network whose IP subnet would collide with another,
+OVN-Kubernetes will report an error to the RouteAdvertisement status.
+
+The frrConfigurationSelector is used in order to determine which FRRConfiguration CR to use for building the OVN-Kubernetes
+driven FRRConfiguration. OVN-Kubernetes needs to leverage a pre-existing FRRConfiguration to be able to find required
+pieces of configuration like BGP peering, etc. If more than one FRRConfiguration is found matching the selector, then
+an error will be propagated to the RouteAdvertisements CR and no configuration shall be done.
+
+##### No Tunnel/Overlay Mode
+
+Changes will be made to the network CRD(s) in order to specify the method of transport for east/west traffic. A new
+field, “transport” will be added with values “geneve” or “none” (default geneve). The transport option in the CRD may be
+used to toggle between using Geneve encapsulation (the default today) or using no encapsulation. With no encapsulation,
+east/west packets are routed directly on the underlay using routing learned via BGP. This is supported for Layer 3
+networks only. Enabling a transport of “none” for Layer 2 networks will have no effect. Enabling a transport of “none”
+without selecting all nodes for RouteAdvertisements will also break east/west traffic between those nodes, as there will
+not be routes propagated into the BGP network for unselected nodes. Users should expect disruption when the transport of
+a network is changed.
```

Comment:
Seems like there should be error checking here rather than "will have no effect" / "will break traffic".